src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSignature.SlhDsa.cs

```diff
@@ -0,0 +1,131 @@
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Security.Cryptography.X509Certificates;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography.Pkcs
+{
+    internal partial class CmsSignature
+    {
+        static partial void PrepareRegistrationSlhDsa(Dictionary<string, CmsSignature> lookup)
+        {
+            lookup.Add(Oids.SlhDsaSha2_128s, new SlhDsaCmsSignature(Oids.SlhDsaSha2_128s));
+            lookup.Add(Oids.SlhDsaShake128s, new SlhDsaCmsSignature(Oids.SlhDsaShake128s));
+            lookup.Add(Oids.SlhDsaSha2_128f, new SlhDsaCmsSignature(Oids.SlhDsaSha2_128f));
+            lookup.Add(Oids.SlhDsaShake128f, new SlhDsaCmsSignature(Oids.SlhDsaShake128f));
+            lookup.Add(Oids.SlhDsaSha2_192s, new SlhDsaCmsSignature(Oids.SlhDsaSha2_192s));
+            lookup.Add(Oids.SlhDsaShake192s, new SlhDsaCmsSignature(Oids.SlhDsaShake192s));
+            lookup.Add(Oids.SlhDsaSha2_192f, new SlhDsaCmsSignature(Oids.SlhDsaSha2_192f));
+            lookup.Add(Oids.SlhDsaShake192f, new SlhDsaCmsSignature(Oids.SlhDsaShake192f));
+            lookup.Add(Oids.SlhDsaSha2_256s, new SlhDsaCmsSignature(Oids.SlhDsaSha2_256s));
+            lookup.Add(Oids.SlhDsaShake256s, new SlhDsaCmsSignature(Oids.SlhDsaShake256s));
+            lookup.Add(Oids.SlhDsaSha2_256f, new SlhDsaCmsSignature(Oids.SlhDsaSha2_256f));
+            lookup.Add(Oids.SlhDsaShake256f, new SlhDsaCmsSignature(Oids.SlhDsaShake256f));
+        }
+
+        private sealed class SlhDsaCmsSignature : CmsSignature
+        {
+            private string _signatureAlgorithm;
+
+            internal SlhDsaCmsSignature(string signatureAlgorithm)
+            {
+                _signatureAlgorithm = signatureAlgorithm;
+            }
+
+            protected override bool VerifyKeyType(object key) => key is SlhDsa;
+            internal override bool NeedsHashedMessage => false;
+
+            internal override RSASignaturePadding? SignaturePadding => null;
+
+            internal override bool VerifySignature(
+#if NET || NETSTANDARD2_1
+                ReadOnlySpan<byte> valueHash,
+                ReadOnlyMemory<byte> signature,
+#else
+                byte[] valueHash,
+                byte[] signature,
+#endif
+                string? digestAlgorithmOid,
+                ReadOnlyMemory<byte>? signatureParameters,
+                X509Certificate2 certificate)
+            {
+                if (signatureParameters.HasValue)
+                {
+                    throw new CryptographicException(
+                        SR.Format(SR.Cryptography_UnknownAlgorithmIdentifier, _signatureAlgorithm));
+                }
+
+                // The spec (as of May 5, 2025) has strength requirements on the hash, but we will
+                // not enforce them here. If the callers wants to enforce them, they can do so by themselves.
+
+                SlhDsa? publicKey = certificate.GetSlhDsaPublicKey();
```

Comment:
We didn't get set a good example in the other algorithms, but this should be disposed (The classic-crypto algs can be cleaned up in a separate PR. Maybe we should make an issue for that?)